app/src/main/java/com.example.weatherapp/data/dialog/WeatherDayListener.kt

```diff
@@ -0,0 +1,5 @@
+package com.example.weatherapp.data.dialog
+
+interface WeatherDayListener {
```

Comment:
as this listener is used in the adapter and for the adapter you can add it in the same file with it

Reply:
Removed a listener interface and added a function callback to WeatherAdapter.